Source/ProgressionSystemRuntime/Private/Data/PSSaveGameData.cpp

```diff
@@ -105,18 +105,7 @@ float UPSSaveGameData::GetProgressionReward(EEndGameState EndGameState)
 {
 	// Verify that the current row exists in the map to prevent creating a new entry
 	const FPSRowData& CurrentProgressionSettingsRowData = UPSWorldSubsystem::Get().GetCurrentProgressionSettingsRowByName();
-
-	switch (EndGameState)
-	{
-	case EEndGameState::Win:
-		return CurrentProgressionSettingsRowData.WinReward * UPSWorldSubsystem::Get().GetDifficultyMultiplier();
-	case EEndGameState::Draw:
-		return CurrentProgressionSettingsRowData.DrawReward * UPSWorldSubsystem::Get().GetDifficultyMultiplier();
-	case EEndGameState::Lose:
-		return CurrentProgressionSettingsRowData.LossReward * UPSWorldSubsystem::Get().GetDifficultyMultiplier();
-	default:
-		return 0.f; // Return a default reward of 0.f if the row does not exist
-	}
+	return *CurrentProgressionSettingsRowData.ProgressionEndGameState.Find(EndGameState) * UPSWorldSubsystem::Get().GetDifficultyMultiplier();
```

Comment:
I see next problems here:
- Dereferencing the pointer directly can lead to crashes if the EndGameState is not found in the progression data. Handle this by checking if the pointer is null and defaulting to 0.f. Now, It will crash for every game state that game entered, but it was not added to your progression data table.
- Readability and debugging: Avoid doing calculations in a single line, which makes debugging harder. Breaking down the expression improves readability.

### Example
```cpp
constexpr float DefaultMultiplier = 0.f;
const float DifficultyMultiplier = UPSWorldSubsystem::Get().GetDifficultyMultiplier();
const FPSRowData& CurrentProgressionSettingsRowData = UPSWorldSubsystem::Get().GetCurrentProgressionSettingsRowByName();

const float* FoundValue = CurrentProgressionSettingsRowData.ProgressionEndGameState.Find(EndGameState);
const float FinalValue = FoundValue ? *FoundValue : DefaultMultiplier;

return FinalValue * DifficultyMultiplier;
```

Reply:
done: https://github.com/h4rdmol/ProgressionSystem/pull/23/commits/5551b24ec239146c839c896999a0ced66a742d4c